arduino_unit_tests/cc1120_ext_regs.h

```diff
@@ -0,0 +1,134 @@
+#define FREQ0               0x0E
+#define IF_ADC2             0x0F
+#define IF_ADC1             0x10
+#define IF_ADC0             0x11
+#define FS_DIG1             0x12
+#define FS_DIG0             0x13
+#define FS_CAL3             0x14
+#define FS_CAL2             0x15
+#define FS_CAL1             0x16
+#define FS_CAL0             0x17
+#define FS_CHP              0x18
+#define FS_DIVTWO           0x19
+#define FS_DSM1             0x1A
+#define FS_DSM0             0x1B
+#define FS_DVC1             0x1C
+#define FS_DVC0             0x1D
+#define FS_LBI              0x1E
+#define FS_PFD              0x1F
+#define FS_PRE              0x20
+#define FS_REG_DIV_CML      0x21
+#define FS_SPARE            0x22
+#define FS_VCO4             0x23
+#define FS_VCO3             0x24
+#define FS_VCO2             0x25
+#define FS_VCO1             0x26
+#define FS_VCO0             0x27
+#define GBIAS6              0x28
+#define GBIAS5              0x29
+#define GBIAS4              0x2A
+#define GBIAS3              0x2B
+#define GBIAS2              0x2C
+#define GBIAS1              0x2D
+#define GBIAS0              0x2E
+#define IFAMP               0x2F
+#define LNA                 0x30
+#define RXMIX               0x31
+#define XOSC5               0x32
+#define XOSC4               0x33
+#define XOSC3               0x34
+#define XOSC2               0x35
+#define XOSC1               0x36
+#define XOSC0               0x37
+#define ANALOG_SPARE        0x38
+#define PA_CFG3             0x39
+
+#define WOR_TIME1           0x64
+#define WOR_TIME0           0x65
+#define WOR_CAPTURE1        0x66
+#define WOR_CAPTURE0        0x67
+#define BIST                0x68
+#define DCFILTOFFSET_I1     0x69
+#define DCFILTOFFSET_I0     0x6A
+#define DCFILTOFFSET_Q1     0x6B
+#define DCFILTOFFSET_Q0     0x6C
+#define IQIE_I1             0x6D
+#define IQIE_I0             0x6E
+#define IQIE_Q1             0x6F
+#define IQIE_Q0             0x70
+#define RSSI1               0x71
+#define RSSI0               0x72
+#define MARCSTATE           0x73
+#define LQI_VAL             0x74
+#define PQT_SYNC_ERR        0x75
+#define DEM_STATUS          0x76
+#define FREQOFF_EST1        0x77
+#define FREQOFF_EST0        0x78
+#define AGC_GAIN3           0x79
+#define AGC_GAIN2           0x7A
+#define AGC_GAIN1           0x7B
+#define AGC_GAIN0           0x7C
+#define CFM_RX_DATA_OUT     0x7D
+#define CFM_TX_DATA_IN      0x7E
+#define ASK_SOFT_RX_DATA    0x7F
+#define RNDGEN              0x80
+#define MAGN2               0x81
+#define MAGN1               0x82
+#define MAGN0               0x83
+#define ANG1                0x84
+#define ANG0                0x85
+#define CHFILT_I2           0x86
+#define CHFILT_I1           0x87
+#define CHFILT_I0           0x88
+#define CHFILT_Q2           0x89
+#define CHFILT_Q1           0x8A
+#define CHFILT_Q0           0x8B
+#define GPIO_STATUS         0x8C
+#define FSCAL_CTRL          0x8D
+#define PHASE_ADJUST        0x8E
+#define PARTNUMBER          0x8F
```

Comment:
You should also define the actual value in the PARTNUMBER register. Based on the datasheet, it should be 0x48 for the CC1120. Reading from the PARTNUMBER register and checking if it's equal to the actual part number is also a good first test to run.

Reply:
This is kind of defined, at least for the registers not in the extended address space, within the cc1120_unit_tests.cpp file. Though I agree, it would be best to put this in a .h somewhere instead. Thing is, I wanted to put it in an array, and doing initializations like that doesn't seem to work unless I put it in a function scope.